CHANGELOG.md

```diff
@@ -10,6 +10,18 @@ internal API changes are not present.
 Main (unreleased)
 -----------------
 
+### Bugfixes
+
+- Upgrade `otelcol` components from OpenTelemetry v0.126.0 to v0.128.0 (@korniltsev, @dehaansa)
+  - [`kafkaexporter`]: Allow kafka exporter to produce to topics based on metadata key values (#39208)
+  - [`kafkareceiver`]: enforce a backoff mechanism on non-permanent errors, such as when the queue is full (#39580)  
+  - [`kafkareceiver`]: Don't restart the Kafka consumer on failed errors when message marking is enabled for them. (#39909)
+  - [`datadogexporter`]: Fix automatic intial point dropping when converting cumulative monotonic sum metrics (#40426)
+  - [`datadogexporter`]: config `tls::insecure_skip_verify` is now taken into account in metrics path. (#40507, #40085)
+  - [`datadogexporter`]: Correctly treat summary counts as cumulative monotonic sums instead of cumulative non-monotonic sums (#40176)
```

Comment:
```suggestion
  - [`datadogexporter`]: Correctly treat summary counts as cumulative monotonic sums instead of cumulative non-monotonic sums
  https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/40176
```
I think GitHub will automatically try to link those to Alloy PRs.